### DISCUSSION TITLE
soup errors when upgrading from 2.3.40 to 2.3.70

### DISCUSSION THREAD

**Discussion by innovate-support** [Security-Onion-Solutions/securityonion#5437] [Unsupported Versions] [answered]
I have 1 Manager and 3 Forward nodes.  When running soup on the Manager it runs through all it's upgrade processes and appears to be successfully updating, but then it throws this error.
![error1](https://user-images.githubusercontent.com/20424476/132542081-c828a36e-ebe6-4703-a02e-f30ab5f4ab2e.png)

The logs weren't helpful.  They just said the same thing without further details.
![log1](https://user-images.githubusercontent.com/20424476/132542801-21212418-0a53-4810-830d-0f001a34a6c6.png)

Running soup again resulted in another 'unhandled error'.  But this time the logs contained a possible clue.  There seems to be a problem with the Hive database.
![log2](https://user-images.githubusercontent.com/20424476/132543526-c71133c4-4e52-45c1-868e-d1ac1e7cf648.png)

Running soup one more time says everything is complete.
![soup complete](https://user-images.githubusercontent.com/20424476/132544385-37a0e265-e394-41df-b03d-76ca1cc37195.png)

The problem is that even though it says it's done, something isn't quite right with the Hive.  The Manager gui shows v2.3.70 and it even pushes the upgrade to the Sensors successfully, but Alert Escalations don't work anymore.  This was posted in
https://github.com/Security-Onion-Solutions/securityonion/discussions/5295

Rebooting everything didn't fix anything either.

Is there perhaps a cleanup or database halting step that should be performed on the Hive dbase prior to running soup?

**Comment by dougburks** [comment]
I just tested an upgrade from 2.3.40 to 2.3.70 and it worked for me:
```
### soup has been served at Fri Sep 10 11:14:06 UTC 2021 ###
```

After your first run of `soup`, I see that you ran `sudo tail /root/soup.log`. Did you just look at the tail output or did you review the entire log for actual errors?

**Comment by innovate-support** [comment]
Trying the upgrade again today.  As I watch the repos pushing I'm seeing numerous notifications of
**received unexpected HTTP status: 500 Internal Server Error**
for various docker containers.  Most retry and seems to succeed and proceeds.  But not 'thehive'.  Those fail.  

![Screenshot from 2021-09-13 12-20-20](https://user-images.githubusercontent.com/20424476/133136655-3f7d2ad1-382a-427e-acda-257bdb99f830.png)

**Comment by innovate-support** [comment, marked as answer]
The upgrade seems to be ok on the Manager this time, but the Sensors are having issues now.  They all show the new version number in the Grid webpage, but they're all listed in a 'Fault' state.  Running so-status on them shows numerous containers labeled as MISSING.  
![Screenshot from 2021-09-13 15-39-36](https://user-images.githubusercontent.com/20424476/133160196-75b1da16-eb18-49b4-b22f-df7a63628803.png)

I then ran **`sudo so-docker-refresh`** and it pushed a lot of containers even though I just ran soup a few hours ago.  After I waited a bit for salt to do it's propogations the Sensors came back up and showed everything OK again.
